README_vi.md

```diff
@@ -0,0 +1,1647 @@
+ *
+ # Except this file
+ !.gitignore
+```
+
+Một quy ước chung khác là tạo một tệp trống trong thư mục có tiêu đề .gitkeep.
+
+
+```sh
+$ mkdir mydir
+$ touch mydir/.gitkeep
+```
+
+Bạn cũng có thể đặt tên tệp là .keep, trong trường hợp dòng thứ hai ở trên sẽ ```touch mydir/.keep```
+
+### Tôi muốn cache một username và password cho một repository
+
+Bạn có thể có một repository yêu cầu xác thực.  Trong trường hợp này bạn có thể cache một username và password vì vậy bạn không phải nhập nó vào mỗi lần push / pull. Việc xác thực có thể làm điều này cho bạn. 
+
+```sh
+$ git config --global credential.helper cache
+# Set git to use the credential memory cache
+```
+
+```sh
+$ git config --global credential.helper 'cache --timeout=3600'
+# Set the cache to timeout after 1 hour (setting is in seconds)
+```
+
+### Tôi muốn làm cho Git bỏ qua các quyền và thay đổi về filemode
+
+```sh
+$ git config core.fileMode false
+```
+
+Nếu bạn muốn đặt hành vi này là hành vi mặc định cho người dùng đã đăng nhập, thì hãy sử dụng:
+
+```sh
+$ git config --global core.fileMode false
+```
+
+### Tôi muốn đặt user toàn cục
+
+Để cấu hình thông tin người dùng được sử dụng trên tất cả các repository cục bộ và để đặt tên có thể nhận dạng khi xem lại phiên bản lịch sử:
+
+```sh
+$ git config --global user.name “[firstname lastname]”
+```
+Để đặt địa chỉ email sẽ được liên kết với mỗi điểm đánh dấu lịch sử:
+
+```sh
+git config --global user.email “[valid-email]”
+```
+
+### Tôi muốn thêm màu cho command Git
+
+Để thiết lập màu command tự động cho Git để dễ dàng xem lại:
+
+```sh
+$ git config --global color.ui auto
+```
+
+## Tôi không nghĩ mình đã làm gì sai
+
+Vì vậy, bạn đang say - bạn `reset` vài thứ, hoặc bạn merge sai nhánh, hoặc bạn force push và bây giờ bạn không thể tìm thấy các commit của bạn. Bạn biết, tại một số thời điểm, bạn đã làm tốt, và bạn muốn quay trở lại trạng thái bạn đang ở đó.
+
+Đây là những gì `git reflog` cho. `reflog` theo dõi bất kỳ thay đổi nào đối với mẹo của nhánh, ngay cả khi mẹo đó không được tham chiếu bởi nhánh hoặc tag. Về cơ bản, mỗi lần HEAD thay đổi, một mục mới được thêm vào reflog. Điều này chỉ hoạt động đối với các repository cục bộ, thật đáng buồn, và nó chỉ theo dõi các chuyển động (ví dụ: không thay đổi một tệp không được ghi ở bất kỳ đâu).
+
+```sh
+(master)$ git reflog
+0a2e358 HEAD@{0}: reset: moving to HEAD~2
+0254ea7 HEAD@{1}: checkout: moving from 2.2 to master
+c10f740 HEAD@{2}: checkout: moving from master to 2.2
+```
+
+Các reflog ở trên cho thấy một checkout từ master đến nhánh 2.2 và trở lại. Từ đó, có một thiết lập cứng để một commit cũ hơn. Hoạt động mới nhất được thể hiện ở đầu được gắn nhãn `HEAD@{0}`.
+
+Nếu nó chỉ ra rằng bạn vô tình di chuyển trở lại, các reflog sẽ chứa commit master chỉ đến (0254ea7) trước khi bạn vô tình giảm 2 commit
+
+If it turns out that you accidentally moved back, the reflog will contain the commit master pointed to (0254ea7) before you accidentally dropped 2 commits.
```

Comment:
Missed a line. :)

Reply:
I forgot to delete this line. I fixed this by delete this line